### ISSUE TITLE
Button bar disappears

### ISSUE THREAD

**Issue by twifty** [ajayyy/SponsorBlock#316]
This is not so much a problem with SB, but the stylesheets used by YT and our manipulation of the DOM.

My own plugin adds 3 buttons to the toolbar (capture, download and mutate). SB in its default form has 2 (open and starts-now ). Clicking start will add the clear, clicking stop will add the submit; giving a total of 4 buttons.

These 7 buttons are too much for the all mighty google's youtube to handle. The whole ".ytp-right-controls" bar will flow to the next row, with parent having overflow hidden, it disappears from view.

No fault of a single extension, but combined...

My mutate button mentioned above offers 4 additional buttons (rotate L/R, flip V/H). Also, since I deem it not very important, I've appended rather then prepended. These 4 buttons are diplayed in a popout when the main button is hovered. I mention this because I think a solution to the disappearing toolbar is to add the rightmost buttons to a dropdown/dropup (with last visible button icon in the bar being a simple triangle which can be hovered).

Thoughts, ideas?

Any extension which adds buttons is going to face the same issue. Adding too many buttons breaks the bar for every extension. We need an extension/developer friendly solution.

I propose both of our extensions add a fix, but to be friendly, a data attribute should be added to the ".ytp-right-controls" element which would signify to a later loading extension that the problem has been fixed.

**Comment by ajayyy** [comment]
Maybe there is a way to detect the number of height in lines?

**Comment by ajayyy** [comment]
Full screen also grants more space. Embeds can have almost no space

**Comment by Joe-Dowd** [comment]
I think if this was going to be done, making an npm library to handle the player buttons which both extensions use would be the way to go.

**Comment by Joe-Dowd** [comment]
The issue could already be there for us on embeds anyway as there's no way to know what size the player is going to be in certain conditions.

**Comment by ajayyy** [comment]
I have seen it appear on embeds that are too small

**Comment by ajayyy** [comment]
Change the width on https://youtubevideoembed.com/#! to see the issue. The right controls move down to the next line to an area that is invisible

**Comment by ajayyy** [comment]
![image](https://user-images.githubusercontent.com/12688112/78510397-cbec9000-7762-11ea-8f03-cb6bde0a40ae.png)

This is what happens when you scroll down

**Comment by Joe-Dowd** [comment]
Yeah, I say we fix it using a library that others can use to solve the problem too in a way that doesn't break compatibility.

**Comment by ajayyy** [comment]
https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect could be used. If `document.querySelector(".ytp-left-controls")` and `document.querySelector(".ytp-right-controls")` have different y values, then there is an issue.

**Comment by twifty** [comment]
Why not just attach a `IntersectionObserver` before adding any buttons.

**Comment by ajayyy** [comment]
I'll look into making the library once I have finished my category and voting changes.

And, if you want to talk about specifics @twifty , you can send a message in the [Discord](https://discord.gg/QnmVMpU)
